spec/support/matchers/accessibility.rb

```diff
@@ -135,11 +135,41 @@ def descriptors(element)
   failure_message do |element|
     <<-STR.squish
       Expected element would have computed name "#{name}".
-      Found #{computed_name(element)}.
+      Found #{AccessibleName.new(page:).computed_name(element)}.
     STR
   end
 end
 
+RSpec::Matchers.define :have_unique_form_landmark_labels do
+  # Could maybe be extended to other landmarks by modifying this function
+  def landmarks(page)
+    page.all(:css, 'form').reject { |element| element[:'aria-hidden'] == 'true' }
+  end
+
+  match do |page|
+    labels = landmarks(page).map { |element| AccessibleName.new(page:).computed_name(element) }
+
+    labels.none?(&:nil?) && labels.uniq.length == labels.length
```

Comment:
Do we need the first part of this condition? I think if we were to adapt this for other screens, we're more concerned with testing for unique labels if a page has more than one form, so if the page has only one form and it has a `nil` label, that seems fine?

Reply:
Mmmm. Maybe `(labels.length == 1 || labels.none?(&:nil))` ?

Reply:
Yeah, though that doesn't check that the labels are unique across multiple forms. Originally I was thinking it'd be enough to keep the rest of the current logic (`labels.uniq.length == labels.length`) but that wouldn't flag if there was multiple forms and only one or some of them were labeled.

Maybe this?

```rb
labels.one? || labels.compact.uniq.length == labels.length
```